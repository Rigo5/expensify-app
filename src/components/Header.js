import React from 'react';
import { NavLink } from 'react-router-dom';


const Header = () => {
    return (
        <header>
            <h1>Expensive app</h1>
            <div>
                <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
                <NavLink to="/help" activeClassName="is-active">Help</NavLink>
                <NavLink to="/create" activeClassName="is-active">Create</NavLink>
            </div>
        </header>
    );
}

export default Header;