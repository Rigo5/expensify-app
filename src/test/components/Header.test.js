import ReactShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import Header from  '../../components/Header';
import { shallow } from 'enzyme';

//il test dei componenti non è un vero e proprio test, nel senso che vengono utilizzati gli snapshot.
//Gli snap shot sono sostanzialmente html del componente renderizzato  che vengono storati. 
//Se il componete viene modificato per passare il test, o combacia con quello storato o si approva la modifica e 
//si fa l'update dello snapshot
//Enzyme usa gli snap shot ma ci aggiunge funzionalita di analisi dell albero html per analizzare numero nodi e specifiche di questi.
//Cose che con reacy-renderer non se po fare

test('shoul render the header correctly', () => {
    //renderizziamo il componente a un livello più basso del browser per testarlo
    /* const renderer = new ReactShallowRenderer(); 
    //renderizziamo jsx component
    renderer.render(<Header />); 
    expect(renderer.getRenderOutput()).toMatchSnapshot(); */

    const wrapper = shallow(<Header />);
    //questo permette di selezionare nello snap tramite class, id  o tag name e ritorna array
    //expect(wrapper.find('h1').length).toBe(1)
    //per permettere a enzyme di lavorare con gli snapshot dobbiamo far si che quelli che va a salvare non abbiano 
    //la parte di codice spazzatura ma solo il componente renderizzato dobbiamo installare enzyme-to-json. 
    //Questo è stato impostato nel file di configurazione di jest e gira in automatico
    expect(wrapper).toMatchSnapshot();
})