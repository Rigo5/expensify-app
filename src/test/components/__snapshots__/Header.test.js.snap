// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoul render the header correctly 1`] = `
<header>
  <h1>
    Expensive app
  </h1>
  <div>
    <NavLink
      activeClassName="is-active"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
    <NavLink
      activeClassName="is-active"
      to="/help"
    >
      Help
    </NavLink>
    <NavLink
      activeClassName="is-active"
      to="/create"
    >
      Create
    </NavLink>
  </div>
</header>
`;
